### ISSUE TITLE
[BUG]: bad error report because of assetion error when there is UnusedCliTokensError

### ISSUE THREAD

**Issue by OrHayat** [BrianPugh/cyclopts#103]
code that reproduce the issue - checkd it on existing main branch (v2.3.0)

```python
import cyclopts
from cyclopts import Parameter
from typing import Annotated

app = cyclopts.App()
#same issue happen even if u setup console on app
# from rich.console import Console
# app=cyclopts.App(console=Console())

@app.command(name="store")
def store_item(item: Annotated[tuple[str,str],Parameter(name="--item")]):
    """
    kv:
        key value of item to store
    """
    print(item)

if __name__=="__main__":
    app()
```

input that reproduce the error
```bash
poetry run python3 ./test.py   store --item a  g v
Traceback (most recent call last):
  File "/home/orhayat/cyclopts/cyclopts/core.py", line 622, in parse_args
    raise UnusedCliTokensError(
cyclopts.exceptions.UnusedCliTokensError: Unused Tokens: ['v'].

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/orhayat/cyclopts/./test.py", line 20, in <module>
    app()
  File "/home/orhayat/cyclopts/cyclopts/core.py", line 674, in __call__
    command, bound = self.parse_args(
  File "/home/orhayat/cyclopts/cyclopts/core.py", line 630, in parse_args
    assert e.console
AssertionError
```

it worked  as expected in v2.2.0

```bash
:~/cyclopts (366a036 | v2.2.0) is 📦 v0.0.0 via 🐍 v3.10.6
> poetry run python3 ./test.py   store  --item a b c
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Unused Tokens: ['c'].                                                                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by BrianPugh** [comment]
Hi @OrHayat! Thanks for all the bug requests, I should have an opportunity later today to fix them and cut a new release.

**Comment by BrianPugh** [comment]
This is fixed in v2.3.1. If you believe it's not fully fixed we can re-open the issue. Thanks!
